.github/workflows/check.yaml

```diff
@@ -52,6 +54,16 @@ jobs:
     - name: Run unit tests
       run: tox -e unit
 
+    - name: Determine system architecture
+      run: echo "SYSTEM_ARCH=$(uname -m)" >> $GITHUB_ENV
+    - name: Upload Unit Test Coverage File
+      uses: actions/upload-artifact@v4
+      with:
+        if-no-files-found: ignore
+        name: coverage-unit-${{ env.SYSTEM_ARCH }}
+        path: coverage-unit
```

Comment:
I think this may not be working as expected: since the coverage db file name is the same for all arches. When downloading and merging the artifact files, only one artifact file will remains.

e.g.

uploaded artifacts:
- coverage-unit-aarch64/coverage-unit
- coverage-unit-amd64/coverage-unit
- coverage-func-amd64/coverage-unit
- coverage-func-amd64/coverage-unit

download and merging artifacts (even before `coverage combine`):
- coverage-unit

 Same below

Reply:
I didn't understand your point.
Yeah, they will have the same name for several arch, but when we download it's just for the arch of the runner.

```
      - name: Download Coverage Files
        uses: actions/download-artifact@v4
        with:
          pattern: coverage-*-${{ env.SYSTEM_ARCH }}
```
 
 So if the runner is `aarch64`, it will download just the artifacts of this arch and merge it.

Reply:
Ah I see, but don't we want to download all the coverage files, across all arches, to merge them all? This would get coverage for any arch-specific code.

Reply:
> Ah I see, but don't we want to download all the coverage files, across all arches, to merge them all? This would get coverage for any arch-specific code.

Yes, that's my question. I am not sure how tics workflow work. Do we need to upload multiple times to tiobeweb for different arches?

Reply:
Maybe the arch specific part is not relevant here since it only has amd64 here. I will ask this in https://github.com/canonical/hardware-observer-operator/pull/395  when this is done. Feel free to take a look at the comments above and resolve it

Reply:
I don't see in the SSDLC spec about tics integration what the projects should do for multi arch and the GUI seems to not be prepared for that. Moreover, in our projects most of the time we don't use low-level system operation.

I think we can let to be out of scope by now. If there is a clear guidance in the future, merging multiple coverage from different archs or sending into different files we can easily adapt it.